src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -10411,30 +10412,65 @@ private ErrorPropertySymbol CreateErrorPropertySymbol(ImmutableArray<PropertySym
                         if (m.IsStatic) continue;
                         break;
                 }
+                methods.Add(m);
+            }
+
+            if (!OverloadResolution.FilterMethodsForUniqueSignature(methods, out useParams))
+            {
+                methods.Free();
+                return null;
+            }
+
+            foreach (var m in methods)
+            {
                 if (!isCandidateUnique(ref method, m))
                 {
+                    methods.Free();
                     return null;
                 }
             }
+
             if (node.SearchExtensionMethods)
             {
+                methods.Clear();
                 var receiver = node.ReceiverOpt!;
                 foreach (var scope in new ExtensionMethodScopes(this))
                 {
                     var methodGroup = MethodGroup.GetInstance();
                     PopulateExtensionMethodsFromSingleBinder(scope, methodGroup, node.Syntax, receiver, node.Name, node.TypeArgumentsOpt, BindingDiagnosticBag.Discarded);
                     foreach (var m in methodGroup.Methods)
                     {
-                        if (m.ReduceExtensionMethod(receiver.Type, Compilation) is { } reduced &&
-                            !isCandidateUnique(ref method, reduced))
+                        if (m.ReduceExtensionMethod(receiver.Type, Compilation) is { } reduced)
                         {
-                            methodGroup.Free();
-                            return null;
+                            methods.Add(reduced);
                         }
                     }
                     methodGroup.Free();
+
+                    if (!OverloadResolution.FilterMethodsForUniqueSignature(methods, out bool useParamsForScope))
+                    {
+                        methods.Free();
+                        return null;
+                    }
+
+                    if (methods.Count != 0)
+                    {
+                        useParams = useParamsForScope;
```

Comment:
>useParams = useParamsForScope; [](http://example.com/codeflow?start=24&length=30)

This looks suspicious. How do we want to handle different result from different scopes? What is our intent? Consider stating what the method is expected to return in ```useParams``` under different circumstances.

Reply:
First, I think the `methods` list should be cleared for each scope here, that was an oversight.

The intent is to do the following in each scope:
- lookup the methods,
- filter them via overload resolution
  - also use overload resolution to determine whether `params` should be used for this scope,
  - if there are no applicable candidates in this scope, we don't want to change `useParams` (because the overload resolution helper won't return anything interesting for its `out bool useParams` parameter since there are no candidates) - that's the reason for the `if (methods.Count != 0)`,
  - now I realize we should also add (with tests): check if `useParams` changes across the scopes and the non-extension method result - if it does, we don't have a unique signature
- if the signature isn't unique, return early,
- otherwise continue to the next scope.

Reply:
> that's the reason for the ```if (methods.Count != 0)```

Consider refactor the logic and not call ```FilterMethodsForUniqueSignature``` if we have no methods to filter. I assume we cannot get an empty result from ```FilterMethodsForUniqueSignature``` when we give it a non-empty input.